task-13/src/main/java/com/epam/winter_java_lab/task_13/service/MessageService.java

```diff
@@ -2,136 +2,78 @@
-        this.messageRepo = messageRepo;
->>>>>>> 04818681b3ffc775807441cf756a5b5e07d1c8ed
         this.messageTransformer = messageTransformer;
     }
 
     @Transactional(readOnly = true)
     public Page<MessageDto> findAllMessages(Pageable pageable) {
-<<<<<<< HEAD
+
         return messageRepository.findAll(pageable)
                 .map(messageTransformer::transformToDto);
-=======
-        return messageRepo.findAll(pageable)
-                .map(messageTransformer::transform);
->>>>>>> 04818681b3ffc775807441cf756a5b5e07d1c8ed
     }
 
     @Transactional(readOnly = true)
     public Page<MessageDto> getAllMessagesByUserId(Long userId, Pageable pageable) {
-<<<<<<< HEAD
+
         return messageRepository.findMessagesByUserId(userId, pageable)
                 .map(messageTransformer::transformToDto);
-=======
-        return messageRepo.findMessagesByUserId(userId, pageable)
-                .map(messageTransformer::transform);
->>>>>>> 04818681b3ffc775807441cf756a5b5e07d1c8ed
     }
 
     @Transactional(readOnly = true)
     public Page<MessageDto> findMessagesByFilter(String filter, Pageable pageable) {
-        Page<MessageDto> messages;
-        if (!filter.isEmpty()) {
-<<<<<<< HEAD
-            messages = messageRepository.findByTag(filter, pageable)
-                    .map(messageTransformer::transformToDto);
-=======
-            messages = messageRepo.findByTag(filter, pageable)
-                    .map(messageTransformer::transform);
->>>>>>> 04818681b3ffc775807441cf756a5b5e07d1c8ed
-        } else {
-            messages = findAllMessages(pageable);
-        }
-        return messages;
-    }
 
-    @Transactional(readOnly = true)
-<<<<<<< HEAD
-    public Page<MessageDto> getMessagesByCreatedDateTime(Pageable pageable) {
-        return messageRepository.findTop10ByOrderByCreatedDateTimeDesc(pageable)
+        return messageRepository.findByTagLike(filter, pageable)
                 .map(messageTransformer::transformToDto);
     }
 
+    @Transactional(readOnly = true)
+    public Page<MessageDto> getMessagesByCreatedDateTime() {
+
+    return messageRepository.findMessageTop(PageRequest.of(PAGE_NUMBER,SIZE))
+            .map(messageTransformer::transformToDto);
+}
+
     public void deleteMessageById(Long messageId) {
+
         messageRepository.deleteById(messageId);
     }
 
     public void saveMessage(MessageDto messageDto) {
+
         messageRepository.save(messageTransformer.transformToEntity(messageDto));
     }
 
     public void editMessage(MessageDto messageDto) {
         Message message = messageTransformer.transformToEntity(messageDto);
-        message.setText(messageDto.getText());
-        message.setTag(messageDto.getTag());
 
         saveMessage(messageTransformer.transformToDto(message));
     }
 
-    public MessageDto findMessage(Long id)  {
+    public MessageDto findMessage(Long id) throws NotFoundException {
```

Comment:
We don't have to notify users of our api about runtume exceptions. so this throws section is redundant